challenger.go

```diff
@@ -19,14 +30,22 @@ type InvoiceRequestGenerator func(price int64) (*lnrpc.Invoice, error)
 type LndChallenger struct {
 	client        lnrpc.LightningClient
 	genInvoiceReq InvoiceRequestGenerator
+
+	invoiceStates  map[lntypes.Hash]lnrpc.Invoice_InvoiceState
```

Comment:
Hmm, I guess it would take a _ton_ of invoices for this to really become an issue memory wise. In the future though, we should definitely limit the number of entries that we'll accumulate here.